src/gui/DatabaseWidget.cpp

```diff
@@ -473,26 +472,39 @@ void DatabaseWidget::deleteEntries()
 
     // get all entry pointers as the indexes change when removing multiple entries
     QList<Entry*> selectedEntries;
+    QMap<Entry*, int> referencesMap;
     for (const QModelIndex& index : selected) {
-        selectedEntries.append(m_entryView->entryFromIndex(index));
+        Entry* entry = m_entryView->entryFromIndex(index);
+        selectedEntries.append(entry);
+        QList<Entry*> references = m_db->resolveReferences(entry->uuid());
+        for (auto const& el : selectedEntries)
+            references.removeAll(el);
+        referencesMap.insert(entry, references.size());
```

Comment:
Instead of using a referencesMap and lumping everything into one message below, you should use this same loop to process the delete function for each selected entry. That way a user can choose to delete or not delete a particular selected entry "in the moment" and not have to cancel the whole operation and deselect a particular one. If there are no entries with references, no message should be shown beyond the "Are you sure you want to delete XX entries?"

Reply:
What should be the options available for references? `Replace`, `Ignore`, `Keep original`, `Cancel`? Depending on selected set of options, flow might get complicated as we will need to track underlying dependencies between every field that might be removed.

Reply:
Keep it simple: Copy Values, Delete All, Skip

Reply:
`Delete all` is far from simple in implementation terms. 
I went with Replace / Ignore / Skip for now. What do you think?